### PR TITLE
completed express-post-json

### PR DESCRIPTION
**Pull request by Dprosser4**
Screen Shot 2023-01-27 at 13 39 25 PM

### CODE REVIEW

File: express-post-json/index.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+app.get('/api/grades', (req, res) => {
+  const gradesArr = [];
+  for (const key in grades) {
+    gradesArr.push(grades[key]);
+  }
+  res.json(gradesArr);
+});
+
+app.use(express.json());
```
Comment (RobertGardner) [inline express-post-json/index.js:14]: By convention, all the request handling middleware is done before setting up the get/post/delete handlers.

File: express-post-json/index.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+app.post('/api/grades', (req, res) => {
+  if (req.body.name === undefined || req.body.course === undefined || req.body.score === undefined) {
+    res.sendStatus(400);
+  } else {
+    const newGrade = req.body;
+    newGrade.id = nextId;
+    grades[nextId] = newGrade;
+    newGrade.id = nextId;
```
Comment (RobertGardner) [inline express-post-json/index.js:23]: Why do you do this assignment twice?

**Review by RobertGardner — APPROVED** [review]
Minor comments.

**Review by RobertGardner — CHANGES_REQUESTED** [review]
I just realized your solution is incomplete. The spec says "The server should respond with the same grade object it received, plus its `id` using the status code `201`." You only respond with 201.

**Comment by Dprosser4** [comment]
Screen Shot 2023-01-27 at 14 36 06 PM
Screen Shot 2023-01-27 at 14 36 22 PM

**Review by RobertGardner — APPROVED** [review]

**Pull request merged**